src/features/hotspots/checklist/HotspotChecklistCarousel.tsx

```diff
@@ -12,62 +13,97 @@ export type ChecklistItem = {
   complete?: boolean
   showAuto?: boolean
   autoText?: string
-  completeText?: string
-  background?: 1 | 2 | 3 | 4
 }
 
 type Props = {
   checklistData: ChecklistItem[]
+  percentComplete: number
 }
 
-const HotspotChecklistCarousel = ({ checklistData }: Props) => {
-  const firstIndex = checklistData.findIndex((i) => !i.complete)
+const HotspotChecklistCarousel = ({
+  checklistData,
+  percentComplete,
+}: Props) => {
+  const [slideIndex, setSlideIndex] = useState(0)
+  const colors = useColors()
+  const isAndroid = Platform.OS === 'android'
 
   const keyExtractor = useCallback((item: ChecklistItem, index: number) => {
     return `${item.key}.${index}`
   }, [])
 
-  const isAndroid = Platform.OS === 'android'
+  const onScrollEnd = useCallback(
+    (e) => {
+      const windowWidth = Dimensions.get('window').width
+      const pageNum = Math.min(
+        Math.max(
+          Math.floor(e.nativeEvent.contentOffset.x / windowWidth + 0.5),
+          0,
+        ),
+        checklistData.length,
+      )
+      if (slideIndex !== pageNum) {
+        setSlideIndex(pageNum)
+      }
+    },
+    [checklistData.length, slideIndex],
+  )
+
   const renderItem = useCallback(
-    (item: { item: ChecklistItem }) => (
-      <HotspotChecklistItem
-        title={item.item.title}
-        description={item.item.description}
-        complete={item.item.complete}
-        showAuto={item.item.showAuto}
-        autoText={item.item.autoText}
-        completeText={item.item.completeText}
-        background={item.item.background}
-        isAndroid={isAndroid}
-      />
-    ),
-    [isAndroid],
+    (item: { index: number; item: ChecklistItem }) => {
```

Comment:
I think this would be easier to read if it were something like...
```ts
  type ListItem = { index: number; item: ChecklistItem }
  const renderItem = useCallback(
    ({ index, item }: ListItem) => {
      return (
        <HotspotChecklistItem
          index={index}
          title={item.title}
          description={item.description}
          complete={item.complete}
          showAuto={item.showAuto}
          autoText={item.autoText}
          itemKey={item.key}
          percentComplete={percentComplete}
          isAndroid={isAndroid}
          totalCount={checklistData.length}
        />
      )
    },
    [isAndroid, percentComplete, checklistData.length],
  )
```

Reply:
I like it, updated